src/command_modules/azure-cli-role/tests/test_role.py

```diff
@@ -8,73 +8,112 @@
 import os
 import tempfile
 import time
+import unittest
 
+from azure.cli.testsdk import \
+    (LiveTest, ResourceGroupPreparer, KeyVaultPreparer, JMESPathCheck as JMESPathCheckV2)
+import azure.cli.core.azlogging as azlogging
 from azure.cli.testsdk.vcr_test_base import VCRTestBase, JMESPathCheck, \
     ResourceGroupVCRTestBase, NoneCheck, MOCKED_SUBSCRIPTION_ID
 
+logger = azlogging.get_az_logger(__name__)
 
-class RbacSPSecretScenarioTest(ResourceGroupVCRTestBase):
-    def __init__(self, test_method):
-        super(RbacSPSecretScenarioTest, self).__init__(__file__, test_method)
-        self.resource_group = 'cli_create_rbac_sp'
-        self.sp_name = 'http://' + self.resource_group
 
-    def test_create_for_rbac_with_secret(self):
-        if self.playback:
-            return  # live-only test, will enable playback once resolve #635
-        self.execute()
+class RbacSPSecretScenarioTest(LiveTest):
 
-    def body(self):
-        subscription_id = self.cmd('account list --query "[?isDefault].id" -o tsv')
-        scope = '/subscriptions/{}'.format(subscription_id)
-        self.cmd(
-            'ad sp create-for-rbac -n {0} --scopes {1} {1}/resourceGroups/{2}'.format(self.sp_name,
-                                                                                      scope,
-                                                                                      self.resource_group),
-            checks=[
-                JMESPathCheck('name', self.sp_name)
-            ])
-        self.cmd('ad sp create-for-rbac -n {}'.format(self.sp_name),
-                 allowed_exceptions="'{}' already exists.".format(self.sp_name))
-        self.cmd('role assignment list --assignee {} --scope {}'.format(self.sp_name, scope),
-                 checks=[JMESPathCheck("length([])", 1)])
-        self.cmd(
-            'role assignment list --assignee {} -g {}'.format(self.sp_name, self.resource_group),
-            checks=[JMESPathCheck("length([])", 1)])
-        self.cmd(
-            'role assignment delete --assignee {} -g {}'.format(self.sp_name, self.resource_group),
-            checks=NoneCheck())
-        self.cmd('role assignment delete --assignee {}'.format(self.sp_name), checks=NoneCheck())
-        self.cmd('ad app delete --id {}'.format(self.sp_name))
+    @ResourceGroupPreparer(name_prefix='cli_create_rbac_sp_minimal')
+    def test_create_for_rbac_with_secret(self, resource_group):
 
+        sp_name = 'http://{}'.format(resource_group)
+        self.cmd('ad sp create-for-rbac -n {}2'.format(sp_name))
```

Comment:
Please add a few verification, at least the `name` part, or maybe just remove it since we have similar testing below